src/zod-extensions.ts

```diff
@@ -48,13 +48,15 @@ export function extendZodWithOpenApi(zod: typeof z) {
     });
 
     if (isZodType(this, 'ZodObject')) {
-      const initialExtend = this.extend;
+      const originalExtend = this.extend;
 
       result.extend = function (...args: any) {
-        const extendedResult = initialExtend.apply(result, args);
+        const extendedResult = originalExtend.apply(this, args);
```

Comment:
Hmmm I can't remember why I did this. Can you explain what the difference is? What was the unexpected behavior beforehand?

Reply:
Should be pretty much the same. The difference is in this code:

```ts
const a = z.object({ a: 1 })
const b = z.object({ b: 1 })

b.extend = a.extend

b.extend({ c: 1 })
// Before: { a: 1, c: 1 }
// After: { b: 1, c: 1 }
```

Since the method was defined on the prototype, the second behavior is the more correct one since it was relying on `this` before and the method wasn't bound.

Reply:
Actually was it defined on the prototype hm, let me check

Reply:
Ok so it looks like the `originalExtend` is already bound so my change won't change anything. Still, in the other case - if it wasn't bound, then the original definition would change behavior while the new one won't.